spritzle/tests/test_resume_data.py

```diff
@@ -0,0 +1,69 @@
+#
+# test_resume_data.py
+#
+# Copyright (C) 2016 Andrew Resch <andrewresch@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.    If not, write to:
+#   The Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor
+#   Boston, MA    02110-1301, USA.
+#
+
+import asyncio
+import asynctest
+from pathlib import Path
+import shutil
+
+import libtorrent as lt
+
+resume_data_path = Path(__file__).resolve().parent / 'resource' / 'resume_data'
+torrent_dir = Path(__file__).resolve().parent / 'resource' / 'torrents'
+
+
+async def test_load(core):
```

Comment:
Could we create a different fixture for these tests if we don't want the core to be started or stopped by the fixture?

Reply:
Something like this.. we'd probably have to move the settings out to a module level attribute so we can pass it in when call core.start() in tests using core_nostart (not sold on this name) fixture. Maybe we should make Core take a settings dict in it's __init__ instead of start().

```python
 def core_init():
     config = Config(in_memory=True, config_dir='/tmp')
     state_dir = Path(tempfile.mkdtemp(prefix='spritzle-test'))
     return Core(config, state_dir)

@pytest.fixture(scope='function')
def core_nostart(loop):
    yield core_init()

@pytest.fixture(scope='function')
def core(loop):
    core = core_init()
    settings = {
        'enable_upnp': False,
        'enable_natpmp': False,
        'enable_lsd': False,
        'enable_dht': False,
        'anonymous_mode': True,
        'alert_mask': 0,
        'stop_tracker_timeout': 0,
    }
    with patch.object(core, 'start', partial(core.start, settings=settings)):
        yield core
    if core.session is not None:
        loop.run_until_complete(core.stop())
    shutil.rmtree(str(state_dir))
```

Reply:
If stopping is important, I think it might be okay the way it is. The stopping will happen if you forget/something goes wrong. It won't start automatically, but if you are using `app` (and starting it), or `cli` fixtures it will start and stop automatically. If you aren't using those other fixtures, chances are you wanted to control that manually anyway.

Reply:
Oh right, because we're using the app it calls the on_shutdown handler which calls core.stop()?

Reply:
I still don't quite understand why it's ok to do this though because in your tests you are calling core.start() after it's already started.. this is something that you shouldn't do as I don't think I ever intended start() to be idempotent and it could have strange side-effects in the future.

We should only call start() and stop() once per test.